romeo/src/state.rs

```diff
@@ -469,13 +473,44 @@ impl State {
 			State::Initialized {
 				deposits,
 				withdrawals,
+				stacks_block_height,
 				..
 			} => {
 				let deposit_tasks = deposits.iter_mut().filter_map(|deposit| {
 					match deposit.mint.as_mut() {
 						None => {
-							deposit.mint = Some(TransactionRequest::Created);
-							Some(Task::CreateMint(deposit.info.clone()))
+							// We often receive the deposit before the
+							// transaction is actually mined. By scheduling the
+							// transaction for a block later than the current
+							// one we make ourselves resilient to mining delays
+							// without complex logic.
+							let scheduled_block_height = *stacks_block_height
+								+ MINT_TRANSACTION_DELAY_BLOCKS;
+							deposit.mint =
+								Some(TransactionRequest::Scheduled {
+									block_height: scheduled_block_height,
+								});
+							info!("Scheduled deposit {} for minting on stacks block height {}.",
+								deposit.info.txid, scheduled_block_height);
+							None
+						}
+						Some(TransactionRequest::Scheduled {
+							block_height,
+						}) => {
+							if stacks_block_height.ge(&block_height) {
+								// Only initiate the mint task if the current
+								// stacks block is or is after the stacks block
+								// for which the mint is scheduled.
+								deposit.mint =
+									Some(TransactionRequest::Created);
+								info!(
+									"Created mint for {}.",
+									deposit.info.txid
+								);
+								Some(Task::CreateMint(deposit.info.clone()))
+							} else {
+								None
+							}
```

Comment:
Consider flattening the matching pattern.

```suggestion
}) if stacks_block_height.ge(&block_height) => {
// Only initiate the mint task if the current
// stacks block is or is after the stacks block
// for which the mint is scheduled.
    deposit.mint = Some(TransactionRequest::Created);info!("Created mint for {}.", deposit.info.txid);
    Some(Task::CreateMint(deposit.info.clone()))
```

Reply:
As much as I love this, clippy and fmt will hate it.

Reply:
In fact, this what I originally had, except you accidentally deleted the `None` part.

Reply:
Really? https://github.com/stacks-network/sbtc/actions/runs/6538570444/job/17754800452?pr=293

Reply:
Get rekt! And, the part I didn't realize here was the additional pattern match here is what you were talking about:

```rust
Some(TransactionRequest::Scheduled {
	block_height, 
}) if (block_height <= stacks_block_height) => { // <------ Part of the pattern matching! Cool!
...
}
```

Thanks Carlos, this is much better, 10/10.